File: 5/5-1.pl
{
    local $/=undef;
    open FILE, "input.txt";
    $input = <FILE>;
    close FILE;
}

my $nice = 0;

foreach my $line (split("\n", $input)) {
    if ($line =~ /(ab|cd|pq|xy)/) {
        next;
    }
    my $count = ($line =~ tr/[aeiou]//);
    if ($count < 3) {
        next;
    }
    if ($line !~ /(.)\1/) {
        next;
    }
    $nice++;
}

print $nice."\n";

File: 5/5-1.rb
FORBIDDEN = /(ab|cd|pq|xy)/
VOWELS = /[aeiou]/
REPEATED = /(.)\1/

nice = []
naughty = []

def nice? string
  !!((string =~ FORBIDDEN).nil? && string.scan(VOWELS).length >= 3 && string =~ REPEATED)
end

File.read("input.txt").chomp.split("\n").each do |line|
  nice?(line) ? nice << line : naughty << line
end

puts nice.length
